ansible/roles/atmo-gui-tweaks/tasks/desktop-background.yml

```diff
@@ -2,27 +2,33 @@
 - name: Copy desktop background to instance
   copy:
     src: "{{ background_img }}"
-    dest: /usr/share/backgrounds/cloud_background.jpg
+    dest: '{{ background_dest }}'
 
-- name: Change desktop background on CentOS
-  become: yes
-  become_user: "{{ ATMOUSERNAME }}"
-  command: >
-    gconftool-2 -s -t string /desktop/gnome/background/picture_filename "/usr/share/backgrounds/cloud_background.jpg"
-  when: ansible_distribution == "CentOS"
+- name: check if XFCE is desktop environment
+  stat:
+    path: '/usr/bin/xfce4-session'
+  register: xfce
 
-# Note: These tasks trick the os by replacing their defaults with our own instead of properly
-# setting the background because "Unable to autolaunch a dbus-daemon without a $DISPLAY for X11"
-- name: Change desktop background on Ubuntu
-  copy:
-    src: /usr/share/backgrounds/cloud_background.jpg
-    dest: /usr/share/backgrounds/default.jpg
-    remote_src: True
-  when: ansible_distribution == "Ubuntu" and ansible_distribution_major_version < "16"
+- name: desktop environment is XFCE
+  set_fact:
+    change_background_cmd: 'xfconf-query -c xfce4-desktop --property /backdrop/screen0/monitor0/workspace0/last-image --create --type string --set'
+  when: xfce.stat.exists
 
-- name: Change desktop background on Ubuntu 16+
-  copy:
-    src: /usr/share/backgrounds/cloud_background.jpg
-    dest: /usr/share/backgrounds/xfce/xfce-teal.jpg
-    remote_src: True
-  when: ansible_distribution == "Ubuntu" and ansible_distribution_major_version >= "16"
+- name: check if Gnome is desktop environment
+  stat:
+    path: '/usr/bin/gnome-session'
+  register: gnome
+
+- name: desktop environment is Gnome
+  set_fact:
+    change_background_cmd: 'gconftool-2 -s -t string /desktop/gnome/background/picture_filename'
+  when: gnome.stat.exists and not xfce.stat.exists
+
+- name: Change desktop background
+  become: yes
+  become_user: '{{ ATMOUSERNAME }}'
+  shell: 'export $(dbus-launch); DISPLAY=:{{ item }} {{ change_background_cmd }} "{{ background_dest }}"'
```

Comment:
I believe you can remove the `with_items` here. If you wouldn't mind throwing a comment in to say why dbus-launch is required. If DISPLAY must be set i think we can just use `:0`.